user-declaration-util/src/main/java/org/sunbird/migration/RecordProcessor.java

```diff
@@ -0,0 +1,219 @@
+  }
+
+  /**
+   * this method should be used to get the instance of the class..
+   *
+   * @param connectionFactory
+   * @param requestParams
+   * @return
+   */
+  public static RecordProcessor getInstance(
+      ConnectionFactory connectionFactory, RequestParams requestParams) {
+    return new RecordProcessor(connectionFactory, requestParams);
+  }
+
+  /**
+   * this method is responsible to fetch the user external id data from cassandra and the table name
+   * is usr_external_identity and after fetching will convert the resultSet to List<User>
+   *
+   * @return List<User>
+   */
+  private List<UserDeclareEntity> getUserDataFromDbAsList() {
+    ResultSet resultSet = connection.getRecords("select * from user_declarations");
+    List<UserDeclareEntity> usersList = CassandraHelper.getUserListFromResultSet(resultSet);
+    logger.debug(usersList.iterator().next());
+    return usersList;
+  }
+
+  /**
+   * this method will be used to process externalId this method will get the count of total number
+   * of records that need to be processed
+   *
+   * @return
+   */
+  public void startProcessingUserDeclarations() throws IOException {
+    List<UserDeclareEntity> userSelfDeclareLists = getRecordsToBeProcessed();
+    logger.info("Update SelfDeclare Lists");
+    updateUserSelfDeclareLists(userSelfDeclareLists);
+    int count[] = new int[1];
+    logger.info(
+        "==================Starting Processing Self Declared Users=======================================");
+    userSelfDeclareLists
+        .stream()
+        .forEach(
+            userSelfDeclareObject -> {
+              try {
+                startTracingRecord(userSelfDeclareObject.getUserId());
+                boolean isSuccess = performSequentialOperationOnRecord(userSelfDeclareObject);
+                if (isSuccess) {
+                  count[0] += 1;
+                }
+              } catch (Exception ex) {
+                logExceptionOnProcessingRecord(
+                    userSelfDeclareObject.getUserId(),
+                    userSelfDeclareObject.getOrgId(),
+                    userSelfDeclareObject.getPersona());
+              } finally {
+                endTracingRecord(userSelfDeclareObject.getUserId());
+              }
+            });
+    logger.info(
+        "==================Endingrting Processing Self Declared Users=======================================");
+
+    // Close self declare file writer
+    closeFwSelfDeclaredFailureWriterConnection();
+    closeFwSelfDeclaredSuccessWriterConnection();
+
+    connection.closeConnection();
+    logger.info(
+        "Total Records: "
+            + userSelfDeclareLists.size()
+            + " Successfully Self Declared Users Migrated: "
+            + count[0]
+            + " Total Self Declared Users Migration Failed:"
+            + (userSelfDeclareLists.size() - count[0]));
+  }
+
+  private void updateUserSelfDeclareLists(List<UserDeclareEntity> userSelfDeclareLists) {
```

Comment:
Pls add a comment that you're filtering out all declarations which are intact - i.e., those that don't contain CSV offending characters.

Reply:
ok